src/index.ts

```diff
@@ -1,27 +1,35 @@
 import 'source-map-support/register';
-import {createServer} from 'http';
-import { Config, createApp } from '@foal/core';
+import { Config, ServiceManager, createApp } from '@foal/core';
 import { AppController } from './app/controllers/app.controller';
 import type { Application } from 'express';
+import * as http from 'http';
+import { WebsocketService } from './app/services/ws.service';
+
 
 async function main() {
-  const app = (await createApp(AppController)) as Application;
+
+  const serviceManager = new ServiceManager();
+  
+  const app = (await createApp(AppController, {serviceManager})) as Application;
 
   const socket = Config.get('listen_socket', 'string');
-  const httpServer = createServer(app);
+  const httpServer = http.createServer(app);
+  await serviceManager.get(WebsocketService).attachHttpServer(httpServer);
+
   if (socket) {
     httpServer.listen(socket, () => {
       console.log(`Listening on socket ${socket}...`);
+      
     });
   } else {
     const port = Config.getOrThrow('listen_port', 'number');
     httpServer.listen(port, () => {
-      console.log(`Listening on port ${port}...`);
+      console.log(`Listening on port ${port}`);
     });
   }
 }
 
 main().catch(err => {
   console.error(err instanceof Error ? err.stack : err);
   process.exit(1);
-});
+});
```

Comment:
```suggestion
});

```
Nit: Ideally we keep EOF newlines consistent. :) You might want to run the linter to make sure your contributions are formatted consistently with the rest of the codebase, but to be honest I don't remember if I left the codebase in a state where the linter passed everywhere...